routes/api/v1/forecast.js

```diff
@@ -0,0 +1,189 @@
+var express = require("express");
+var router = express.Router();
+var User = require('../../../models').User;
+var axios = require('axios');
+
+var get_location = function(location){
+  return axios.get(`https://maps.googleapis.com/maps/api/geocode/json?key=${process.env.GOOGLE_API_KEY}&address=${location}`)
+  .then(response => {
+    return response.data.results[0].geometry.location;
+  })
+  .catch(error => {
+    return error;
+  });
+}
+
+var get_weather = function(location){
+  return axios.get(`https://api.darksky.net/forecast/${process.env.DARK_SKY_API_KEY}/${location}`)
+  .then(response => {
+    return response.data;
+  })
+  .catch(error => {
+    return error;
+  });
+}
+
+var parseWeather = function(location,forecast){
+  var current_forecast = forecast["currently"]
+  var hourly_header = forecast["hourly"]
+  var todays_forecast = forecast["daily"]["data"][0]
+  var hourly_forecast = forecast["hourly"]["data"]
+  var daily_forecast = forecast["daily"]["data"]
+
+  var parsed_current_forecast = { location: location,
+    currently: {
+      time: current_forecast["time"],
+      glance_summary: current_forecast["summary"],
+      icon: current_forecast["icon"],
+      current_temperature: current_forecast["temperature"],
+      feels_like_temp: current_forecast["apparentTemperature"],
+      humidity: current_forecast["humidity"],
+      visibility: current_forecast["visibility"],
+      uv_index: current_forecast["uvIndex"],
+      high_temp: todays_forecast["temperatureHigh"],
+      low_temp: todays_forecast["temperatureLow"],
+      today_summary: hourly_header["summary"]
+    },
+    hourly: [{
```

Comment:
One thing I found with the hourly & daily forecast that may simplify your setup a little bit, is that you can iterate over the array with a forEach function, something along the lines of: 

forecast["hourly"]["data"].forEach(hourlyWeather);

function hourlyWeather(hour) {
 // build your hash in here
}

I noticed you're only taking the even indices for hourly though, so that may be a moot point if you're  excluding those for a specific reason. Just a thought on how to cut down some lines of code!

Reply:
That was a stylistic choice when doing the mod3 final. I was operating under the assumption that returning every other hour for the next 12 would be better than the next 6 hours, then I basically just grabbed that code block and converted the symbols to strings for JS and added a couple more entries. 

I've added an issue to my project board to go back and do it that way if I get the last 3 endpoints done.